spec/lib/rbac/filterer_spec.rb

```diff
@@ -1317,17 +1317,29 @@ def get_rbac_results_for_and_expect_objects(klass, expected_objects)
       end
 
       context "with VMs and Templates" do
+        let(:root) do
+          datacenters = FactoryBot.create(:ems_folder, :name => "Datacenters")
+          datacenters.parent = @ems
+          datacenters
+        end
+
+        let(:dc) do
+          datacenter = FactoryBot.create(:ems_folder, :name => "Datacenter1")
+          datacenter.parent = root
+          datacenter
+        end
+
+        let(:hfolder) do
+          hfolder = FactoryBot.create(:ems_folder, :name => "host")
```

Comment:
ASIDE: have you tried:

`FactoryBot.create(:ems_folder, :name => "host", :parent => dc)`

or if that doesn't work,

`FactoryBot.create(:ems_folder, :name => "host").tap { |hfolder| hfolder.parent = dc }`

Reply:
that's right 👍  I used the `tap` version. thanks!